go/runfiles/manifest.go

```diff
@@ -97,10 +107,119 @@ func (m manifest) path(s string) (string, error) {
 	// prefixes of path in the manifest.
 	for prefix := s; prefix != ""; prefix, _ = path.Split(prefix) {
 		prefix = strings.TrimSuffix(prefix, "/")
-		if prefixMatch, ok := m[prefix]; ok {
+		if prefixMatch, ok := m.index[prefix]; ok {
 			return prefixMatch + filepath.FromSlash(strings.TrimPrefix(s, prefix)), nil
 		}
 	}
 
 	return "", os.ErrNotExist
 }
+
+func (m *manifest) open(name string) (fs.File, error) {
+	if name != "." {
+		r, err := m.path(name)
+		if err == ErrEmpty {
+			return emptyFile(name), nil
+		} else if err == nil {
+			return os.Open(r)
+		}
+	}
+
```

Comment:
So once we are done with this `if` block, we can conclude that `name == "." || err == os.ErrNotExist` right?

To make it more clear:

```suggestion
	if name != "." {
		r, err := m.path(name)
		if err == ErrEmpty {
			return emptyFile(name), nil
		}
		if err == nil {
			// found
			return os.Open(r)
		}
		// err == os.ErrNotExist, do a full-tree scan
	}

```